src/Analyzers/CSharp/CodeFixes/UseParameterNullChecking/CSharpUseParameterNullCheckingCodeFixProvider.cs

```diff
@@ -60,6 +60,33 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
                         var parameterReferenceSyntax = nullCoalescing.Left;
                         editor.ReplaceNode(nullCoalescing, parameterReferenceSyntax.WithAppendedTrailingTrivia(SyntaxFactory.ElasticMarker));
                         break;
+                    case IfStatementSyntax { Else.Statement: BlockSyntax { Statements: var statementsWithinElse } } ifStatementWithElseBlock:
+                        {
+                            var parent = (BlockSyntax)ifStatementWithElseBlock.Parent!;
```

Comment:
how do you know hte parent is a block?  was that tested on the analyzer side?

Reply:
We only search statements directly contained within the method block body.

https://github.com/dotnet/roslyn/blob/c83e59dc080d619868e9bae7c79a45104b26c6d1/src/Analyzers/CSharp/Analyzers/UseParameterNullChecking/CSharpUseParameterNullCheckingDiagnosticAnalyzer.cs#L142

Reply:
terrific!